libs/datasets/timeseries.py

```diff
@@ -960,8 +960,19 @@ def from_csv(path_or_buf: Union[pathlib.Path, TextIO]) -> "MultiRegionDataset":
         return dataset
 
     @staticmethod
```

Comment:
This is tangential to this PR but I'm noticing that we have `@functools.lru_cache(maxsize=None)` on several functions that call this function but we don't have it here, even though this is the function that dose the real work and so this is the innermost place we could apply the caching and have it provide the most benefit.  As-is, each function that calls this will probably load the full dataset once before it's cached for subsequent uses.

So I'm wondering if we could / should add `@functools.lru_cache(maxsize=None)` here and remove it from all of the callers.

Feel free to punt / ignore since it's not directly related to this PR but it may speed up CI / tests a bit.

Reply:
Ooh, that's a good flag. I think I'll ship this as-is, but will create a follow-up PR to try that out 🙏